File: 2020 December Bronze/daisychains.py
def avg_flower(flowers, petals):
    count = 0

    # Iterate over all possible subarrays
    for i in range(flowers):
        tsum = 0
        for j in range(i, flowers):
            tsum += petals[j]
            length = j - i + 1
            # Check if the average is an integer
            if tsum % length == 0:
                average = tsum // length
                # Check if any flower in the subarray has the average number of petals
                if average in petals[i:j+1]:
                    count += 1
                    #print(f"Subarray ({i}, {j}): petals = {petals[i:j+1]}, average = {average}")

    return count

# Input reading
num_flowers = int(input())
num_petals = [int(petals) for petals in input().split()]

# Function call and output
result = avg_flower(num_flowers, num_petals)
print(result)

File: 2020 December Bronze/daisychains.java
import java.util.*;

public class daisychains{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int numFlowers = in.nextInt();
        in.nextLine();

        Integer[] numPetals = new Integer[numFlowers];
        for(int i = 0; i < numFlowers; i++){
            numPetals[i] = in.nextInt();
        }

        System.out.println(avgFlower(numFlowers, numPetals));

        in.close();
    }

    public static int avgFlower(int flowers, Integer[] petals){
        int count = 0;
        for(int i = 0; i < flowers; i++){
            int tempSum = 0;
            for(int j = i; j < flowers; j++){
                tempSum += petals[j];
                int length = j - i + 1;
                if (tempSum % length == 0){
                    int average = tempSum / length;
                    for(int e = i; e < j+1; e++){
                        if (petals[e] == average){
                            count++;
                            break;
                        }
                    }
                }
            }
        }
        return count;
    }
}